src/ui/server/controllers/insurance/business/company-details/index.ts

```diff
@@ -99,12 +108,57 @@ const postCompaniesHouseSearch = async (req: Request, res: Response) => {
       }),
       ...PAGE_VARIABLES,
       SUMMARY_LIST: summaryList,
-      companiesHouseNumber,
+      submittedValues,
     });
   } catch (error) {
     console.error('Error posting companies house search', { error });
     return res.redirect(ROUTES.PROBLEM_WITH_SERVICE);
   }
 };
 
-export { get, postCompaniesHouseSearch };
+/**
+ * posts company details
+ * validates tradingName fields
+ * @param req
+ * @param res
+ * @returns validation errors if required fields not entered correctly
+ */
+const post = (req: Request, res: Response) => {
+  try {
+    const { body } = req;
+    const submittedValues = {
+      [COMPANY_HOUSE.INPUT]: body[COMPANY_HOUSE.INPUT],
+      [TRADING_NAME]: body[TRADING_NAME],
+    };
```

Comment:
Doesn't seem that we need to do this, just mapping the same fields?

Reply:
It does not work without this, I have tried